drivers/hud_spm_report/app/models/hud_spm_report/fy2023/spm_enrollment.rb

```diff
@@ -0,0 +1,280 @@
+    end
+
+    def self.detail_headers
+      client_columns = ['client_id', 'first_name', 'last_name', 'personal_id', 'data_source_id']
+      hidden_columns = ['id', 'report_instance_id', 'previous_income_benefits_id', 'current_income_benefits_id', 'enrollment_id'] + client_columns
+      columns = client_columns + (column_names - hidden_columns)
+      columns.map do |col|
+        [col, header_label(col)]
+      end.to_h
+    end
+
+    private_class_method def self.header_label(col)
+      case col.to_sym
+      when :client_id
+        'Warehouse Client ID'
+      when :personal_id
+        'HMIS Personal ID'
+      when :data_source_id
+        'Data Source ID'
+      when :los_under_threshold
+        'LOS Under Threshold'
+      when :previous_street_essh
+        'Previous Street ESSH'
+      else
+        col.humanize
+      end
+    end
+
+    private_class_method def self.start_of_homelessness(filter, household_info, enrollment)
+      age = enrollment.client.age_on([filter.start, enrollment.entry_date].max)
+      start_of_homelessness = if age.present? && age <= 17 &&
+        enrollment.entry_date == household_info.entry_date
+        # Inherit start of homelessness from HoH for any household member aged <- 17 if they entered with the HoH
+        household_info.start_of_homelessness
+      else
+        enrollment.date_to_street_essh
+      end
+      # Start of homelessness is never before birth
+      start_of_homelessness = [start_of_homelessness, enrollment.client.dob].max if start_of_homelessness.present? && enrollment.client.dob.present?
+
+      start_of_homelessness
+    end
+
+    private_class_method def self.move_in_date(household_info, enrollment)
+      # Use the client move in date if they are the HoH
+      return enrollment_own_move_in_date(enrollment) if enrollment.head_of_household?
+      # Don't inherit move in date if the client exited before the HoH moved in
+      return enrollment_own_move_in_date(enrollment) if enrollment.exit.present? && household_info.move_in_date.present? && enrollment.exit.exit_date <= household_info.move_in_date
+      # Use the client's entry date if a client entered the household after the HoH had already moved in
+      return enrollment.entry_date if household_info.move_in_date.present? && enrollment.entry_date > household_info.move_in_date
+
+      # Otherwise, inherit move in date from HoH
+      household_info.move_in_date
+    end
+
+    private_class_method def self.eligible_funding?(enrollment, start_date, end_date)
+      enrollment.project.funders.open_between(start_date: start_date, end_date: end_date).any? do |funder|
+        funder.funder.in?([2, 3, 4, 5, 43, 44, 54, 55].map(&:to_s))
+      end
+    end
+
+    private_class_method def self.total_income(income_benefit)
+      income_benefit&.hud_total_monthly_income&.clamp(0..) || 0
+    end
+
+    private_class_method def self.earned_income(income_benefit)
+      (income_benefit&.earned_amount || 0).clamp(0..)
```

Comment:
I may be missing something, but this looks like two ways of ding the same thing.  Can we pick one?